cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/transformers/AbstractTransformerConversion.java

```diff
@@ -109,15 +109,41 @@ protected void addAliases(Identifiable<?> identifiable) {
         if (ptcs != null) {
             for (int  i = 0; i < ptcs.size(); i++) {
                 int index = i + 1;
-                Optional.ofNullable(ptcs.get(i)).ifPresent(ptc -> identifiable.addAlias(ptc, Conversion.CGMES_PREFIX_ALIAS + CgmesNames.PHASE_TAP_CHANGER + index));
+                Optional.ofNullable(ptcs.get(i)).ifPresent(ptc -> identifiable.addAlias(ptc, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.PHASE_TAP_CHANGER + index));
             }
         }
         List<String> rtcs = context.cgmes().ratioTapChangerListForPowerTransformer(identifiable.getId());
         if (rtcs != null) {
             for (int i = 0; i < rtcs.size(); i++) {
                 int index = i + 1;
-                Optional.ofNullable(rtcs.get(i)).ifPresent(rtc -> identifiable.addAlias(rtc, Conversion.CGMES_PREFIX_ALIAS + CgmesNames.RATIO_TAP_CHANGER + index));
+                Optional.ofNullable(rtcs.get(i)).ifPresent(rtc -> identifiable.addAlias(rtc, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.RATIO_TAP_CHANGER + index));
             }
         }
     }
+
+    protected static void addCgmesReferences(Identifiable<?> transformer, TapChanger tc) {
+        if (tc == null || tc.getId() == null) {
+            return;
+        }
+        if (tc.getRegulatingControlId() != null) {
+            transformer.setProperty(cgmesReferenceKey(tc.getId(), "TapChangerControl"), tc.getRegulatingControlId());
+        }
+        if (tc.getType() != null) {
+            transformer.setProperty(cgmesReferenceKey(tc.getId(), "type"), tc.getType());
+        }
+        TapChanger tch = tc.getHiddenCombinedTapChanger();
+        if (tch != null) {
+            // All the tap changers have already been added as aliases,
+            // Through properties we only label which one has been combined and kept hidden
+            transformer.setProperty(cgmesReferenceKey(tc.getId(), "hiddenTapChangerId"), tch.getId());
+            transformer.setProperty(cgmesReferenceKey(tch.getId(), "step"), String.valueOf(tch.getTapPosition()));
+            if (tch.getType() != null) {
+                transformer.setProperty(cgmesReferenceKey(tch.getId(), "type"), tch.getType());
+            }
```

Comment:
Why is it necessary?

Reply:
For phase tap changers, the subtype of tap changer (Linear, non-Linear Asymmetrical, non-Linear Symmetrical) must be stored as a property because the subtype class name is required for proper SSH export. The XML element must be named with the subtype. An example for the MicroGrid test case:

```xml
  <cim:PhaseTapChangerAsymmetrical rdf:about="#_6ebbef67-3061-4236-a6fd-6ccc4595f6c3">
    <cim:TapChanger.step>10</cim:TapChanger.step>
    <cim:TapChanger.controlEnabled>true</cim:TapChanger.controlEnabled>
  </cim:PhaseTapChangerAsymmetrical>
```

In PowSyBl all phase tap changers are tabular, and there is no other way to recover this information from the IIDM `PhaseTapChanger` object, so we decided to save it as a property.

Reply:
Okay! In this case, I wrote it in `SSHExport` file, but in export, the default type should be tabular

Reply:
You are right, I fixed the default value (`PhaseTapChanger` is an abstract class in CGMES, a subtype must be used, and default should be `PhaseTapChangerTabular`, according to IIDM modelling).